### PR TITLE
[GLUTEN-2767][GLUTEN-2834][CH] Fix negative decimal32 return wrong value

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#2767, #2834)

## How was this patch tested?

Test by ut

### CODE REVIEW

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp [hunk 1]
```diff
@@ -95,7 +95,14 @@ static void writeFixedLengthNonNullableValue(
     for (size_t i = 0; i < num_rows; i++)
     {
         size_t row_idx = masks == nullptr ? i : masks->at(i);
-        writer.unsafeWrite(col.column->getDataAt(row_idx), buffer_address + offsets[i] + field_offset);
+
+        if (writer.getWhichDataType().isDecimal32())
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:99]: 这个判断条件外提到for外面吧，不确定for里if else会不会影响编译器优化  


if (类型是decimal32)
{
   for循环
}
else
{
  for循环
}

Comment (loneylee) [reply cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp]: fixed

**Comment by baibaichen** [comment]
LGTM

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**